state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1003,6 +1039,19 @@ pub mod pallet {
 						FlipBuyInterval::<T>::set(interval);
 						Self::deposit_event(Event::<T>::BuyIntervalSet { buy_interval: interval });
 					},
+					PalletConfigUpdate::SetMaxSwapRetryDuration { blocks } => {
+						MaxSwapRetryDurationBlocks::<T>::set(blocks);
+						Self::deposit_event(Event::<T>::MaxSwapRetryDurationSet { blocks });
+					},
+					PalletConfigUpdate::SetMaxSwapRequestDuration { blocks } => {
+						// TODO JAMIE: check that this is correct.
+						ensure!(
+							blocks >= SWAP_DELAY_BLOCKS,
+							Error::<T>::MaxSwapRequestDurationTooShort
+						);
```

Comment:
I think we might also have to ensure that the max duration is greater than the max retry duration (?)

(Might depend on the answers to my other questions)

Reply:
`max retry duration`  is part of FoK which is independent from DCA, so i don't understand why we would have that restriction?